src/BunyanDebugStream.ts

```diff
@@ -349,7 +357,14 @@ class BunyanDebugStream extends Writable {
             }
         }
 
-        const joinedPrefixes = prefixes.length > 0 ? `[${prefixes.join(',')}] ` : '';
+        let joinedPrefixes = '';
+        if (this._showPrefixes && prefixes.length > 0) {
+            if (typeof this._showPrefixes === 'function') {
+                joinedPrefixes = `${this._showPrefixes(prefixes)} `;
```

Comment:
If `_showPrefixes()` throws an error here, we may want to fall back on

```ts
joinedPrefixes = `[${prefixes.join(',')}] `;
```

?

Reply:
This is not in line with your date (showDate) formatting behavior. It has no similar fallbacks AFAIC. I'd rather implement this concern separately.

Reply:
Fair enough!  :)